### PR TITLE
Add volume already in replication check

### PR DESCRIPTION
**Pull request by xxwjj**
**What this PR does / why we need it**:
1. Add "volume already been used in replication" check

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/replication.go [hunk 1]
```diff
@@ -69,6 +69,38 @@ func (this *ReplicationPortal) CreateReplication() {
 		return
 	}
 
+	// check if specified volume has already used in other replication.
```
Comment (xing-yang) [inline pkg/api/replication.go]: s/has already used/has already been used

File: pkg/api/replication.go [hunk 2]
```diff
@@ -69,6 +69,38 @@ func (this *ReplicationPortal) CreateReplication() {
...
+		}
+	}
+	if v != nil {
+		model.HttpError(this.Ctx, http.StatusBadRequest,
+			"specified primary volume(%s) has already been used in replication(%s) ",
+			replication.PrimaryVolumeId, v.Id)
+		return
+	}
+
+	// check if specified volume has already used in other replication.
```
Comment (xing-yang) [inline pkg/api/replication.go]: has already been used

File: pkg/db/drivers/etcd/etcd.go [hunk 3]
```diff
@@ -1853,6 +1853,20 @@ func (c *Client) GetReplication(ctx *c.Context, replicationId string) (*model.Re
...
+	if err != nil {
+		return nil, err
+	}
+
+	for _, r := range replications {
+		if volumeId == r.PrimaryVolumeId || volumeId == r.SecondaryVolumeId {
+			return r, nil
+		}
+	}
+	return nil, model.NewNotFoundError(fmt.Sprintf("can't find spcified replication by volume id %s", volumeId))
```
Comment (xing-yang) [inline pkg/db/drivers/etcd/etcd.go]: s/spcified/specified

**Comment by xing-yang** [comment]
Please rebase.

**Comment by leonwanghui** [comment]
@xxwjj I'm wondering why the code coverage is much lower than before.

File: contrib/connector/fc/linuxfc.go [hunk 4]
```diff
@@ -126,7 +126,7 @@ func (l *linuxfc) flushDeviceIO(device string) error {
 	out, err := exec.Command("/bin/bash", "-c", cmd).CombinedOutput()
 	outString := string(out)
 	if err != nil {
-		errMsg := fmt.Sprintf("Error occurred when get device info when detach volume:", outString, err)
+		errMsg := fmt.Sprintf("Error occurred when get device info when detach volume:%s,%v", outString, err)
```
Comment (leonwanghui) [inline contrib/connector/fc/linuxfc.go]: I guess you missed a blank space.

**Pull request merged**